test/e2e/apps/daemon_restart.go

```diff
@@ -245,12 +245,12 @@ var _ = SIGDescribe("DaemonRestart [Disruptive]", func() {
 			&cache.ListWatch{
 				ListFunc: func(options metav1.ListOptions) (runtime.Object, error) {
 					options.LabelSelector = labelSelector.String()
-					obj, err := f.ClientSet.CoreV1().Pods(ns).List(backgroundCtx, options)
+					obj, err := f.ClientSet.CoreV1().Pods(ns).List(context.Background(), options)
```

Comment:
Why is using the `backgroundCtx` still causing problems?

If the test is done, then these functions should not get called anymore and if they are, failing with "context canceled" is the right response.

Reply:
> Informer context only applies to the Run method, the context on the
> ListWatch only causes that current request is cancelled, but the controller
> will restart them, causing an endless loop of cancel/retries.

Why does the controller not stop in this case? The context given to `Run` is cancelled, it should stop instead of entering this endless loop.

Reply:
ah, no, just changed it for uniformity across the code base, not strong opinion

Reply:
So far I have not been able to reproduce this endless loop problem. Perhaps it depends on timing?

Below you re-introduced `context.Background` in `test/e2e/scheduling/preemption.go`. That is a test that I can run against a local cluster. Instead of falling back to `context.Background`, I went in the other direction: I am removing the `stopCh` and use the `ctx` from Ginkgo for both the functions and `Run`.

```patch
diff --git a/test/e2e/scheduling/preemption.go b/test/e2e/scheduling/preemption.go
index bfb199fcc7c..05cec6912d4 100644
--- a/test/e2e/scheduling/preemption.go
+++ b/test/e2e/scheduling/preemption.go
@@ -623,7 +623,6 @@ var _ = SIGDescribe("SchedulerPreemption [Serial]", func() {
                */
                framework.ConformanceIt("runs ReplicaSets to verify preemption running path", func(ctx context.Context) {
                        podNamesSeen := []int32{0, 0, 0}
-                       stopCh := make(chan struct{})
 
                        // create a pod controller to list/watch pod events from the test framework namespace
                        _, podController := cache.NewInformer(
@@ -652,8 +651,7 @@ var _ = SIGDescribe("SchedulerPreemption [Serial]", func() {
                                        },
                                },
                        )
-                       go podController.Run(stopCh)
-                       defer close(stopCh)
+                       go podController.Run(ctx.Done())
 
                        // prepare three ReplicaSet
                        rsConfs := []pauseRSConfig{
```

That's the same approach as the one taken in this `BeforeEach` and it seems to work fine so far.

Reply:
The CI also seems to be happy with daemon_restart.go now:

https://storage.googleapis.com/k8s-triage/index.html?text=RcManager%20created%2Fdeleted%20pods%20after%20restart%20&test=Controller%20Manager%20should%20not%20create%2Fdelete%20replicas%20across%20restart

Reply:
@aojea : is this the controller loop you mentioned?

https://github.com/kubernetes/kubernetes/blob/789dc88fd5ab50c870e43f2e7b339bc07fe0edd6/staging/src/k8s.io/client-go/tools/cache/reflector.go#L269-L280

Reply:
yep, that keeps restarting right?

but anyway, this works fine, I think that I was overthinking it

/close